Assets/Scripts/PlayerController/PlayerMovement.cs

```diff
@@ -31,6 +31,9 @@ public class PlayerMovement : MonoBehaviour
 
     private PlayerNetworkHandler m_PlayerNetworkHandler;
 
+    private PowerUpsManager m_PowerUps;
+    private VelocityModifier m_VelocityModifier;
```

Comment:
From a design perspective, I feel like `PowerUpsManager` and `VelocityModifier` should depend on `PlayerMovement` instead of the other way around. Could we just have `PowerUpsManager` and `VelocityModifier` use `[RequireComponent(typeof(PlayerMovement))]` and call setters in `PlayerMovement` instead?

Reply:
Will look into this. Yes did would be a lot more sensible. This currently does work on arbitrary players, however, I am unsure if the data is passed on via the network to the screens of other players.